build: update CLI version to 1.7.2

rafaelss95 opened the pull request:
Closes #195.

rafaelss95 commented:
I guess `README.md` needs to be updated. `ng serve` simply doesn't work (without run another script before) (see #227). I had to run `yarn start`, in order to run the `build-themes` before start the server.

Also, there's a lint error due to `HttpModule` and its relatives (like `RequestOptions`, etc...) deprecations.
Currently there are many places using them. Is it preferable to disable this rule (for now) and replace them in another PR?

Can you take a look, @jelbourn? Thanks.

rafaelss95 commented:
@jelbourn In fact, there are no unrelated changes. The proposed changes are a fully update in `@angular/cli`, to be in parity with latest version. 

Look at the PR #143, the `cli` version had been updated from `"@angular/cli": "^1.0.0-beta.30"` to `"@angular/cli": "^1.0.0"`... **but** just the version was updated, completely ignoring the fact that it changed a lot from beta to the final version (file names, structure, etc).

So, what I did here was: I created a fresh new project and compared all the changes, and there are a lot of changes, as you could seen.

rafaelss95 commented:
Well, those tslint rules are the default in the latest `@angular/cli`. I can undo those changes if you want.

jelbourn commented:
Yeah- we don't necessarily use the default tslint rules here, anyway; we'd really want the same rules as the main material2 repo (except for the custom rules, which I don't want to copy-paste).

rafaelss95 commented:
No problem, I undid the change of tslint.

jelbourn commented:
@amcdnl could you do a review pass?

amcdnl commented on src/app/shared/navigation-focus/navigation-focus.ts:
```diff
@@ -14,7 +14,7 @@ export class NavigationFocus implements OnInit {
     clearTimeout(lastTimeoutId);
     // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    lastTimeoutId = setTimeout(() => this.el.nativeElement.focus(), 100);
+    lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus(), 100);
```
Why the `window` here?

rafaelss95 replied:
Not sure, but I guess I got an error when tried to run a specific command.

amcdnl commented on src/app/index.ts:
```diff
@@ -1,2 +1,2 @@
 export * from './material-docs-app';
-export * from './app-module';
+export * from './app.module';
```
@jelbourn prefers the `app-module` format.

rafaelss95 replied:
Really :( I'll change ASAP.

amcdnl commented on tsconfig.json:
```diff
@@ -1,16 +1,26 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
```
Why was this added?

amcdnl commented:
@rafaelss95 - Can you rebase?

rafaelss95 commented:
@amcdnl I've rebased, however now there a lot of errors. Lint (which ones I fixed in this PR) and others coming from `@angular/cdk/tree` and `MatBottomSheet`.

rafaelss95 commented:
Done in #431.
